dsl/src/main/scala/org/http4s/dsl/impl/Path.scala

```diff
@@ -326,6 +326,19 @@ abstract class OptionalQueryParamDecoderMatcher[T: QueryParamDecoder](name: Stri
       .toOption
 }
 
+abstract class DefaultQueryParamDecoderMatcher[T: QueryParamDecoder](name: String, default: T) {
```

Comment:
Would `QueryParamDecoderMatcherWithDefault` be a better name? We're not deciding a default query peram.